### PR TITLE
Make the link to the CoVe paper explicit and add documentation on agent coordination

### PR DESCRIPTION
**Pull request by fjuma**
Fixes https://github.com/wise-agents/wise-agents/issues/339

### CODE REVIEW

File: docs/agent_coordination.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+If the confidence score is less than the threshold, the `PhasedCoordinatorWiseAgent` will
+attempt to rephrase the original query and try to execute the phases again using the
+improved query to try to get to a better final answer. This process will be repeated until
+the desired confidence score is achieved or until the max number of iterations has been reached.
+
+> **_NOTE:_** The `PhasedCoordinatorWiseAgent` makes use of agent names and descriptions along
+> with phase names to group the agents into phases. It's important to provide good descriptions
+> for your agents to indicate their capabilities and dependencies. For example, if an agent
+> depends on another agent, please indicate this in the agent's description.
```
Comment (maeste) [inline docs/agent_coordination.md:100]: Maybe we want to describe in more detail the algorithm/idea behind agent selection for the phases to be sure users can understand the importance of description. Maybe something like that mentioning they are instruction for the LLM building up the phases could help.

**Comment by maeste** [comment]
it is fixing also #212

File: docs/agent_coordination.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Wise Agents Coordination
+
+One of the most important features of an AI multi-agent system is the ability for multiple
+agents to work together to perform a complex task or solve a problem. The agents involved
+in the system might make use of an LLM, but they don't have to, they could also perform
+a task using an external API for example. The agents might also make use of different
+LLM models.
+
```
[the remark this answers is not in the document]
Comment (fjuma) [reply docs/agent_coordination.md:8]: Updated.

File: docs/agent_coordination.md (hunk 1 above)
Comment (fjuma) [reply docs/agent_coordination.md:109]: Updated the `How does a `PhasedCoordinatorWiseAgent` work?` section.

**Review by maeste — APPROVED** [review]

**Pull request merged**